tests/UsersTest.php

```diff
@@ -152,4 +146,40 @@ public function testNormalUserShouldNotBeAbleDisableMFA()
             $this->assertEquals(403, $e->getCode());
         }
     }
+
+    public function testsRemoveUserFromEverywhere()
+    {
+        $organization = $this->client->createOrganization($this->testMaintainerId, ['name' => 'ToRemoveOrg']);
+        $project = $this->client->createProject($organization['id'], ['name' => 'ToRemoveProj']);
+        $email = 'remove' . uniqid() . '@keboola.com';
+        $this->client->addUserToProject($project['id'], ['email' => $email]);
+        $user = $this->client->getUser($email);
+        $this->client->addUserToOrganization($organization['id'], ['email' => $user['email']]);
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['email' => $user['email']]);
+        $this->client->removeUser($email);
+
+        $usersInProject = $this->client->listProjectUsers($project['id']);
+        foreach ($usersInProject as $userInProject) {
+            if ($userInProject['id'] === $user['id']) {
+                $this->fail('User has not been deleted from project');
+            }
+        }
+
+        $usersInOrganization = $this->client->listOrganizationUsers($organization['id']);
+        foreach ($usersInOrganization as $userInOrganization) {
+            if ($userInOrganization['id'] === $user['id']) {
+                $this->fail('User has not been deleted from organization');
+            }
+        }
+
+        $usersInMaintainer = $this->client->listMaintainerMembers($this->testMaintainerId);
+        foreach ($usersInMaintainer as $userInMaintainer) {
+            if ($userInMaintainer['id'] === $user['id']) {
+                $this->fail('User has not been deleted from maintainer');
+            }
+        }
+
+        $this->expectException(ClientException::class);
+        $this->client->getUser($email);
```

Comment:
Tady chybí konec IMHO - mělo by to otestovat, že je ten user anonymizovaný, ne? Z toho důvodu nevím, jestli by nebylo lepší udělat toho usera na začátku "pořádně" (tj. se vším, co se u usera dá vyplnit) - takhle vlastně teď testuješ, jestli se smaže neaktivovaný uživatel... 

Plus bude asi potřeba vymyslet, jak se to bude chovat v návaznosti na pozvánky. Typický case asi bude, že někoho pozveš, ale pak zjistíš třeba, že jsi pozval někoho s tečkou v mailu a potřebuješ bez tečky. Tak ideálně budeš toho člověka chtít úplně smazat... a i jeho invite...

Reply:
Taky jsem tam chtěl nacpat nějaký assert... Ale nepřišel jsem na způsob, co vlastně assertovat. Úspěch testu je, že uživatele se nepodařilo najít a to hází exception. Takže to je pro mě úspěšný assert a taky konec testu.
Určitě tady nechci testovat něco jakože se přidal správně do projektu atd. To vůbec není věc tohohle testu.

Reply:
S tím, že se mají mazat i pozvánky máš podle mě pravdu. Je to tak @ErikZigo ?

Musím se přiznat, že tohle je přesně místo, kde mě štve, že si nikdo nesednul a nevymslel zadání a všechno se dělá za pochodu. To je už trochu moc agilní :-1:

Reply:
Ano při mazání uživatele by se měly smazat i všechny jeho pozvánky a join requesty do projektu.
Brzo k tomu přibudou i pozvánky do organizace a maintainera.

Reply:
No, za mě:

```
assertSame('DELETED', $user->name);
assertSame('DELETED', $user->password);
...
assertNotEquals('DELETED', $user->mfaCode);
....
```
Prostě otestovat, že ten "smazaný" uživatel není úplně smazaný, ale je anonymizovaný.

Reply:
@tomasfejfar pořád mi nedocházelo jak získat uživatele, který ḿá e-mail "DELETED" stejně jako 50 dalších, ale on jde vlastně uživatel získat podle ID :+1: Tohle je ten kousek skládanky, co mi chyběl :) Ok. Dodělám.